pkg/util/helper/workstatus.go

```diff
@@ -73,41 +73,28 @@ func AggregateResourceBindingWorkStatus(
 
 	fullyAppliedCondition := generateFullyAppliedCondition(binding.Spec, aggregatedStatuses)
 
-	err = retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
-		currentBindingStatus := binding.Status.DeepCopy()
-
-		binding.Status.AggregatedStatus = aggregatedStatuses
-		// set binding status with the newest condition
-		meta.SetStatusCondition(&binding.Status.Conditions, fullyAppliedCondition)
-		if reflect.DeepEqual(binding.Status, *currentBindingStatus) {
-			klog.V(4).Infof("New aggregatedStatuses are equal with old resourceBinding(%s/%s) AggregatedStatus, no update required.",
-				binding.Namespace, binding.Name)
-			return nil
-		}
-
-		updateErr := c.Status().Update(context.TODO(), binding)
-		if updateErr == nil {
+	var operationResult controllerutil.OperationResult
+	if err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		operationResult, err = UpdateStatus(context.Background(), c, binding, func() error {
```

Comment:
I don't quite understand, the modification you're currently making doesn't seem to have any fundamental difference from the previous logic?

It seems that both are utilizing `RetryOnConflict()` and `.Status().Update()`.

Reply:
I am not saying the usage of `RetryOnConflict()` or `.Status().Update()` has any problem, but we should get the latest binding before updating status.

Reply:
Could you remind me, what is the core modification of your PR?

Reply:
when the update fails, the latest binding will be obtained immediately, but it will not be retried immediately because it's subject to `retry.RetryOnConflict` back-off.. when retrying again, the latest binding obtained last time may not be the latest data, which will increase the chance of retry failure.
so what I want to do is adjust the order and get the latest binding before each update, like [controllerutil.CreateOrUpdate](https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/controller/controllerutil/controllerutil.go#L276).

Reply:
Ok, I got you.

 The core change you made is placing `c.Get()` before `c.Status().Update()`. 

However, the original logic was that `c.Get()` came after `c.Status().Update()`. 

There are several similar codes of this logic. Could we put `UpdateStatus()` in `util`?

Reply:
> There are several similar codes of this logic.

could you help point it out? thanks:)

Reply:
https://github.com/karmada-io/karmada/blob/master/pkg/controllers/cronfederatedhpa/cronfederatedhpa_job.go#L310-L330

https://github.com/karmada-io/karmada/blob/master/pkg/controllers/federatedresourcequota/federated_resource_quota_status_controller.go#L162-L177


https://github.com/karmada-io/karmada/blob/master/pkg/controllers/mcs/service_import_controller.go#L155-L174


I think the code in these places should also have similar logic.